src/Microsoft.ML.Data/DataView/Transposer.cs

```diff
@@ -1652,98 +1576,61 @@ public ValueGetter<TValue> GetGetter<TValue>(int col)
-        // cursor level, not the cursorable level. Is there some non-horrifying way to unify both, somehow?
-        private sealed class SlotRowCursorShim<T> : SynchronizedCursorBase<ISlotCursor>, IRowCursor
-        {
-            private readonly SchemaImpl _schema;
-
-            public Schema Schema => _schema.AsSchema;
-
-            private sealed class SchemaImpl : ISchema
-            {
-                private readonly SlotRowCursorShim<T> _parent;
-                private readonly VectorType _type;
 
-                private IChannel Ch { get { return _parent.Ch; } }
+                public override ValueGetter<UInt128> GetIdGetter() => GetId;
 
-                public Schema AsSchema { get; }
-
-                public int ColumnCount { get { return 1; } }
-
-                public SchemaImpl(SlotRowCursorShim<T> parent, VectorType slotType)
+                private void GetId(ref UInt128 id)
                 {
-                    Contracts.AssertValue(parent);
-                    _parent = parent;
-                    Ch.AssertValue(slotType);
-                    _type = slotType;
-                    AsSchema = Schema.Create(this);
+                    Ch.Check(_slotCursor.SlotIndex >= 0, "Cannot get ID with cursor in current state.");
+                    id = new UInt128((ulong)_slotCursor.SlotIndex, 0);
                 }
 
-                public ColumnType GetColumnType(int col)
-                {
-                    Ch.CheckParam(col == 0, nameof(col));
-                    return _type;
-                }
+                protected override bool MoveNextCore() => _slotCursor.MoveNext();
+            }
+        }
 
-                public string GetColumnName(int col)
-                {
-                    Ch.CheckParam(col == 0, nameof(col));
-                    return "Waffles";
-                }
+        // REVIEW: This shim class is very similar to the above shim class, except at the
+        // cursor level, not the cursorable level. Is there some non-horrifying way to unify both, somehow?
+        private sealed class SlotRowCursorShim<T> : RootCursorBase
+        {
+            private readonly SlotCursor _slotCursor;
 
-                public bool TryGetColumnIndex(string name, out int col)
-                {
-                    if (name == GetColumnName(0))
-                    {
-                        col = 0;
-                        return true;
-                    }
-                    col = -1;
-                    return false;
-                }
+            public override Schema Schema { get; }
 
-                public IEnumerable<KeyValuePair<string, ColumnType>> GetMetadataTypes(int col)
-                {
-                    Ch.CheckParam(col == 0, nameof(col));
-                    return Enumerable.Empty<KeyValuePair<string, ColumnType>>();
-                }
+            public override long Batch => throw new NotImplementedException();
```

Comment:
>throw new NotImplementedException(); [](start = 42, length = 36)

flagging this one too.

Reply:
Thanks @sfilipi , I believe I fixed that in my last commit (I fixed both this and the other one that you had already commented on at the same time), maybe you are looking at the older commit?